packages/middleware-apply-body-checksum/src/applyMd5BodyChecksumMiddleware.ts

```diff
@@ -30,13 +30,12 @@ export const applyMd5BodyChecksumMiddleware =
           digest = options.streamHasher(options.md5, body);
         }
 
-        request = {
-          ...request,
-          headers: {
-            ...headers,
-            "content-md5": options.base64Encoder(await digest),
-          },
+        const cloned = request.clone();
```

Comment:
This isn't a safe assumption. 

`HttpRequest.isInstance(request)` is incorrectly implemented. It only checks that a request is duck-like an HttpRequest **interface**. The `clone` method only exists on the `HttpRequest` **class**.

It wasn't a good idea naming interfaces and classes the same, even worse that they aren't identical types.